syntax = "proto3";

package caffa.rpc;

option java_multiple_files = true;
option java_package = "org.caffa.rpc";

service App {
  rpc Quit(NullMessage) returns (NullMessage) {}
  rpc GetAppInfo(NullMessage) returns (AppInfoReply) {}
  rpc Ping(NullMessage) returns (NullMessage) {}
}

message NullMessage {}

enum AppType
{
  CONSOLE     = 0x00;
  GRPC_SERVER = 0x01;
  GRPC_CLIENT = 0x02;
  GUI         = 0x04;
  WEB         = 0x08;
}

message AppInfoReply
{
  string name         = 1;
  uint32 type         = 2;
  int32 major_version = 3;
  int32 minor_version = 4;
  int32 patch_version = 5;
}

